vpr/src/draw/search_bar.cpp

```diff
@@ -261,7 +280,33 @@ void highlight_blocks(std::string block_name) {
     ClusterBlockId block_id = ClusterBlockId::INVALID();
     block_id = cluster_ctx.clb_nlist.find_block(block_name);
 
-    if (block_id == ClusterBlockId::INVALID()) return; //name not exist
+    if (block_id == ClusterBlockId::INVALID()){
+        warning_dialog_box("Invalid Block Name");
+        return; //name not exist
+    }
 
     highlight_blocks(block_id); //found block
+}
+
+void warning_dialog_box(const char* message){
+    GObject *main_window; // parent window over which to add the dialog
+    GtkWidget *content_area; // content area of the dialog
+    GtkWidget *label; // label to display a message
+    GtkWidget *dialog;
+    // get a pointer to the main window
+    main_window = application.get_object(application.get_main_window_id().c_str());
+    // create a dialog window modal with no button
+    dialog = gtk_message_dialog_new(GTK_WINDOW(main_window),
+            GTK_DIALOG_DESTROY_WITH_PARENT,
+            GTK_MESSAGE_INFO,
+            GTK_BUTTONS_NONE,
+            "Error");
+    // create a label and attach it to content area of the dialog
+    content_area = gtk_message_dialog_get_message_area(GTK_MESSAGE_DIALOG(dialog));
+    label = gtk_label_new(message);
+    gtk_container_add(GTK_CONTAINER(content_area), label);
+    // show the label & child widget of the dialog
+    gtk_widget_show_all(dialog);
```

Comment:
I'm not familiar with GTK, but do we need to do any clean-up after the widget is closed to avoid issues like memory leaks?

Reply:
I am not sure, I found a destroy function for dialogs in the GTK documentation. I will try that function. But my team used similar code without calling the destroy function in 297 and did pass the valgrind test.

Reply:
> I found a destroy function for dialogs in the GTK documentation. I will try that function.

Sounds good.

Since this code is only active when the dialog box opens, any leak may only occur if the dialog box opens. The ECE297 valgrind test probably never opened dialog boxes so may have missed this case.

@vaughnbetz @samehattia Any thoughts?

Reply:
The destroy function works, but the scope of this dialog is within this function. So If I want to free the dialog and call this destroy function, I have to somehow make the dialog pointer a global variable and free it somewhere else, otherwise if I free it inside this function, the dialog box wouldn't show up at all since it gets destroyed right away. I assume I should free it at the very end of the program. Where's should I free it? I think I might need to extern this dialog variable to other files?

Reply:
I'm not sure, I would have assumed `gtk_widget_show_all()` would wait until the widget is closed (but perhaps that's not how GTK works...).

I'll leave it for @samehattia to comment on the right way to do this.

Reply:
You could put the pointer into the draw_state function, which would let you free it later in free_draw_structs.  Or you could make it file scope in search_bar.cpp and then make a free_search_bar() routine that free_draw_structs calls.  Not sure which is cleaner; they're pretty close.

Should also update the comment at the top of draw.cpp as it is out of date with ezgl.

Reply:
Thank you so much. I found a way using a gtk callback function.

Reply:
Using signals (as Tina did in 52a303c) is the cleanest way to destroy widgets.